src/cgame/cg_rocket_datasource.cpp

```diff
@@ -1137,7 +1153,20 @@ static void CG_Rocket_CleanUpMapList( const char* )
 
 static void CG_Rocket_SetMapListIndex( const char*, int index )
 {
-	rocketInfo.data.mapIndex = index;
+	if ( index >= 0 && index < int( rocketInfo.data.mapList.size() ) )
+	{
+		rocketInfo.data.mapIndex = index;
+		g_defaultMap.Set( rocketInfo.data.mapList[ index ].mapLoadName );
+	}
+}
+
+static int CG_Rocket_GetMapListIndex( const char* )
+{
+	if ( rocketInfo.data.mapList.size() != 0 )
+	{
+		CG_Rocket_BuildMapList( nullptr );
```

Comment:
Build the list only if it is already nonempty? Seems weird. Probably this shouldn't be here at all?

Reply:
Ah no it should be `== 0`, well, I copypasted `CG_Rocket_GetResolutionListIndex` the cargo cult way, so I don't know if that code is needed, but when duplicating, that should be `==`. I'll fix that.

Reply:
Since the code worked even with `!=0` this shouldn't be needed at all.